numpy/core/src/multiarray/datetime.c

```diff
@@ -3746,6 +3746,27 @@ recursive_find_object_datetime64_type(PyObject *obj,
     }
 }
 
+/*
+ * handler function for PyDelta values
+ * which may also be in a 0 dimensional
+ * NumPy array
+ */
+static int
+delta_checker(PyObject *obj, PyArray_DatetimeMetaData *meta)
```

Comment:
Why does this function not touch `obj`?

Reply:
The function is nothing more than a container for the code I abstracted from the original function because it is now duplicated.

I think a lot of the datetime handling happens at a metadata level & can indeed just remove the `obj` argument here I think.